core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,14 @@ public XContentBuilder percentageField(String rawFieldName, String readableField
         return this;
     }
 
+    public XContentBuilder nanosAsMillis(String rawFieldName, String readableFieldName, long nanos) throws IOException {
```

Comment:
Ok then we should stick with the current api, we are already printing out nanoseconds in a few places, I found for instance `running_time` in `TaskInfo`. We should be consistent with how we do things elsewhere. That means that the human readable time in the profile api will lose some precision, unless we want to go change how we print out nanoseconds at some point later.